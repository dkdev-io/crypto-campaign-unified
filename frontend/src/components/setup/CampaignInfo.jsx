import React from 'react';

const CampaignInfo = ({ formData, updateFormData, onNext, onPrev }) => {
  return (
    <div>
      <h2>Campaign Information</h2>
      <div className="form-group">
        <label>Campaign Name</label>
        <input 
          className="form-input"
          value={formData.campaignName || ''}
          onChange={(e) => updateFormData({ campaignName: e.target.value })}
        />
      </div>
      <div className="form-group">
        <label>Website</label>
        <input 
          className="form-input"
          value={formData.website || ''}
          onChange={(e) => updateFormData({ website: e.target.value })}
        />
      </div>
      <div className="form-actions">
        <button className="btn btn-secondary" onClick={onPrev}>Back</button>
        <button className="btn btn-primary" onClick={onNext}>Next</button>
      </div>
    </div>
  );
};

export default CampaignInfo;
