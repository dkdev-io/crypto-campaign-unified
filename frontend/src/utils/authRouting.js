/**
 * Utility functions for determining correct authentication routes
 * based on user context and current location
 */

/**
 * Determines the appropriate auth route based on current location
 * @param {string} currentPath - Current pathname from useLocation
 * @param {object} state - Optional state to pass to auth route
 * @returns {object} - Object with pathname and state for navigation
 */
export const getAuthRoute = (currentPath, state = {}) => {
  // If we're in a campaign context, use campaign auth
  if (currentPath?.startsWith('/campaigns') || currentPath?.startsWith('/setup')) {
    return {
      pathname: '/campaigns/auth',
      state: { 
        from: currentPath,
        ...state 
      }
    }
  }

  // If we're in a donor context, use donor auth
  if (currentPath?.startsWith('/donors')) {
    return {
      pathname: '/donors/auth',
      state: { 
        from: currentPath,
        ...state 
      }
    }
  }

  // If we're in an admin context, use admin auth
  if (currentPath?.startsWith('/admin')) {
    return {
      pathname: '/admin/login',
      state: { 
        from: currentPath,
        ...state 
      }
    }
  }

  // For the home page or unknown contexts, check referrer or default to campaigns
  // This handles cases where users land on the root page and need auth
  const referrer = document.referrer
  if (referrer?.includes('/donors')) {
    return {
      pathname: '/donors/auth',
      state: { 
        from: currentPath,
        ...state 
      }
    }
  }

  // Default to campaign auth for root, about, or other general pages
  // since campaigns are the primary use case
  return {
    pathname: '/campaigns/auth',
    state: { 
      from: currentPath,
      ...state 
    }
  }
}

/**
 * Determines auth context from various indicators
 * @param {string} currentPath - Current pathname
 * @param {object} user - Current user object (if any)
 * @returns {string} - 'campaign', 'donor', 'admin', or 'default'
 */
export const getAuthContext = (currentPath, user = null) => {
  // Check URL path first
  if (currentPath?.startsWith('/campaigns') || currentPath?.startsWith('/setup')) {
    return 'campaign'
  }
  
  if (currentPath?.startsWith('/donors')) {
    return 'donor'
  }
  
  if (currentPath?.startsWith('/admin')) {
    return 'admin'
  }

  // Check user type if available
  if (user) {
    if (user.user_metadata?.role === 'admin') {
      return 'admin'
    }
    if (user.app_metadata?.isDonor) {
      return 'donor'
    }
    // Default user type is campaign
    return 'campaign'
  }

  // Default context for unauthenticated users
  return 'campaign'
}