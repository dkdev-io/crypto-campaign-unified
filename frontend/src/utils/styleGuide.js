/**
 * Style Guide Utility Functions
 * Extracts and applies campaign style guide data from Supabase
 */

/**
 * Extract campaign styles with fallbacks
 * @param {Object} campaignData - Campaign data from Supabase
 * @returns {Object} Extracted style configuration
 */
export const extractCampaignStyles = (campaignData) => {
  if (!campaignData) {
    return getDefaultStyles();
  }

  // Priority order: applied_styles > custom_styles > theme_color > defaults
  const appliedStyles = campaignData.applied_styles;
  const customStyles = campaignData.custom_styles;
  const themeColor = campaignData.theme_color;

  return {
    colors: {
      primary:
        appliedStyles?.colors?.primary || customStyles?.colors?.primary || themeColor || '#2a2a72',
      secondary: appliedStyles?.colors?.secondary || customStyles?.colors?.secondary || '#666666',
      accent: appliedStyles?.colors?.accent || customStyles?.colors?.accent || '#28a745',
      background:
        appliedStyles?.colors?.background || customStyles?.colors?.background || '#ffffff',
      text: appliedStyles?.colors?.text || customStyles?.colors?.text || '#333333',
    },
    fonts: {
      heading: {
        family:
          appliedStyles?.fonts?.heading?.suggested ||
          customStyles?.fonts?.heading?.family ||
          'Inter, system-ui, -apple-system, sans-serif',
        weight:
          appliedStyles?.fonts?.heading?.weight || customStyles?.fonts?.heading?.weight || '600',
        size: appliedStyles?.fonts?.heading?.size || customStyles?.fonts?.heading?.size || '1.5rem',
      },
      body: {
        family:
          appliedStyles?.fonts?.body?.suggested ||
          customStyles?.fonts?.body?.family ||
          'Inter, system-ui, -apple-system, sans-serif',
        weight: appliedStyles?.fonts?.body?.weight || customStyles?.fonts?.body?.weight || '400',
        size: appliedStyles?.fonts?.body?.size || customStyles?.fonts?.body?.size || '1rem',
      },
      button: {
        family:
          appliedStyles?.fonts?.button?.suggested ||
          customStyles?.fonts?.button?.family ||
          appliedStyles?.fonts?.heading?.suggested ||
          'Inter, system-ui, -apple-system, sans-serif',
        weight:
          appliedStyles?.fonts?.button?.weight || customStyles?.fonts?.button?.weight || '500',
        size: appliedStyles?.fonts?.button?.size || customStyles?.fonts?.button?.size || '1rem',
      },
    },
    layout: {
      borderRadius:
        appliedStyles?.layout?.recommendations?.borderRadius ||
        customStyles?.layout?.borderRadius ||
        '8px',
      spacing:
        appliedStyles?.layout?.recommendations?.margin || customStyles?.layout?.spacing || '1rem',
    },
    // Source information for debugging
    source: {
      hasAppliedStyles: !!appliedStyles,
      hasCustomStyles: !!customStyles,
      hasThemeColor: !!themeColor,
      websiteAnalyzed: campaignData.website_analyzed,
      stylesApplied: campaignData.styles_applied,
    },
  };
};

/**
 * Get default styles when no campaign data is available
 * @returns {Object} Default style configuration
 */
export const getDefaultStyles = () => ({
  colors: {
    primary: '#2a2a72',
    secondary: '#666666',
    accent: '#28a745',
    background: '#ffffff',
    text: '#333333',
  },
  fonts: {
    heading: {
      family: 'Inter, system-ui, -apple-system, sans-serif',
      weight: '600',
      size: '1.5rem',
    },
    body: {
      family: 'Inter, system-ui, -apple-system, sans-serif',
      weight: '400',
      size: '1rem',
    },
    button: {
      family: 'Inter, system-ui, -apple-system, sans-serif',
      weight: '500',
      size: '1rem',
    },
  },
  layout: {
    borderRadius: '8px',
    spacing: '1rem',
  },
  source: {
    hasAppliedStyles: false,
    hasCustomStyles: false,
    hasThemeColor: false,
    websiteAnalyzed: null,
    stylesApplied: false,
  },
});

/**
 * Generate CSS custom properties from campaign styles
 * @param {Object} campaignData - Campaign data from Supabase
 * @returns {Object} CSS custom properties object
 */
export const generateCSSProperties = (campaignData) => {
  const styles = extractCampaignStyles(campaignData);

  return {
    '--campaign-color-primary': styles.colors.primary,
    '--campaign-color-secondary': styles.colors.secondary,
    '--campaign-color-accent': styles.colors.accent,
    '--campaign-color-background': styles.colors.background,
    '--campaign-color-text': styles.colors.text,
    '--campaign-font-heading': styles.fonts.heading.family,
    '--campaign-font-body': styles.fonts.body.family,
    '--campaign-font-button': styles.fonts.button.family,
    '--campaign-border-radius': styles.layout.borderRadius,
    '--campaign-spacing': styles.layout.spacing,
  };
};

/**
 * Apply campaign styles to a React component's style prop
 * @param {Object} campaignData - Campaign data from Supabase
 * @param {Object} baseStyles - Base styles to merge with
 * @returns {Object} Combined styles object
 */
export const applyCampaignStyles = (campaignData, baseStyles = {}) => {
  const styles = extractCampaignStyles(campaignData);
  const cssProps = generateCSSProperties(campaignData);

  return {
    ...baseStyles,
    ...cssProps,
    // Common style applications
    '--primary-color': styles.colors.primary,
    '--text-color': styles.colors.text,
    '--font-family': styles.fonts.body.family,
  };
};

/**
 * Get button styles based on campaign theme
 * @param {Object} campaignData - Campaign data from Supabase
 * @param {string} variant - Button variant ('primary', 'secondary', 'accent')
 * @returns {Object} Button style object
 */
export const getCampaignButtonStyles = (campaignData, variant = 'primary') => {
  const styles = extractCampaignStyles(campaignData);

  const baseButton = {
    fontFamily: styles.fonts.button.family,
    fontWeight: styles.fonts.button.weight,
    fontSize: styles.fonts.button.size,
    borderRadius: styles.layout.borderRadius,
    padding: `calc(${styles.layout.spacing} * 0.75) calc(${styles.layout.spacing} * 1.5)`,
    border: 'none',
    cursor: 'pointer',
    transition: 'all 0.2s ease-in-out',
  };

  switch (variant) {
    case 'primary':
      return {
        ...baseButton,
        backgroundColor: styles.colors.primary,
        color: '#ffffff',
      };
    case 'secondary':
      return {
        ...baseButton,
        backgroundColor: styles.colors.secondary,
        color: '#ffffff',
      };
    case 'accent':
      return {
        ...baseButton,
        backgroundColor: styles.colors.accent,
        color: '#ffffff',
      };
    default:
      return baseButton;
  }
};

/**
 * Debug function to log style guide information
 * @param {Object} campaignData - Campaign data from Supabase
 */
export const debugCampaignStyles = (campaignData) => {
  const styles = extractCampaignStyles(campaignData);

  console.log('ðŸŽ¨ Campaign Style Debug:', {
    campaignId: campaignData?.id,
    source: styles.source,
    extractedStyles: {
      primaryColor: styles.colors.primary,
      headingFont: styles.fonts.heading.family,
      bodyFont: styles.fonts.body.family,
    },
    rawData: {
      applied_styles: campaignData?.applied_styles,
      custom_styles: campaignData?.custom_styles,
      theme_color: campaignData?.theme_color,
    },
  });

  return styles;
};
