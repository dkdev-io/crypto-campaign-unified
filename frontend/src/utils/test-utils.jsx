import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { vi } from 'vitest';

// Test data factories
export const createMockCampaign = (overrides = {}) => ({
  id: 'test-campaign-1',
  campaign_name: 'Test Campaign',
  email: 'test@example.com',
  candidate_name: 'John Doe',
  theme_color: '#2a2a72',
  suggested_amounts: [25, 50, 100, 250],
  max_donation_limit: 3300,
  website: 'https://test.com',
  wallet_address: '0x123...',
  created_at: '2024-01-01T00:00:00.000Z',
  ...overrides
});

export const createMockDonorData = (overrides = {}) => ({
  fullName: 'Jane Smith',
  email: 'jane@example.com',
  phone: '555-123-4567',
  street: '123 Main St',
  city: 'Anytown',
  state: 'CA',
  zip: '12345',
  employer: 'Test Corp',
  occupation: 'Engineer',
  amount: '100',
  ...overrides
});

export const createMockWalletInfo = (overrides = {}) => ({
  isConnected: true,
  account: '0x742d35Cc6C4C5e5D0E9dF5F9B6A5e8a2c1B9D8e7',
  balance: '1.5678',
  network: {
    name: 'Ethereum Mainnet',
    chainId: '0x1'
  },
  contributorInfo: {
    isKYCVerified: true,
    cumulativeAmount: 0.5,
    remainingCapacity: 1.0
  },
  ...overrides
});

export const createMockFormSubmission = (overrides = {}) => ({
  id: 'submission-1',
  campaign_id: 'test-campaign-1',
  donor_full_name: 'Jane Smith',
  donor_email: 'jane@example.com',
  amount_usd: 100,
  transaction_hash: '0xabc123...',
  status: 'confirmed',
  created_at: '2024-01-01T12:00:00.000Z',
  ...overrides
});

// Custom render function with providers
export const render = (ui, {
  route = '/',
  ...renderOptions
} = {}) => {
  // Set initial URL for router
  window.history.pushState({}, 'Test page', route);
  
  const Wrapper = ({ children }) => (
    <BrowserRouter>
      {children}
    </BrowserRouter>
  );

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// Utility to wait for async operations
export const waitForAsync = () => new Promise(resolve => setTimeout(resolve, 0));

// Mock event creators
export const createMockEvent = (overrides = {}) => ({
  preventDefault: vi.fn(),
  stopPropagation: vi.fn(),
  target: {
    value: 'test-value',
    name: 'test-input',
    checked: false,
    ...overrides.target
  },
  currentTarget: {
    value: 'test-value',
    ...overrides.currentTarget
  },
  ...overrides
});

export const createChangeEvent = (value, name = 'test-input') => 
  createMockEvent({
    target: { value, name },
    currentTarget: { value }
  });

export const createClickEvent = (overrides = {}) =>
  createMockEvent({
    type: 'click',
    ...overrides
  });

export const createSubmitEvent = (overrides = {}) =>
  createMockEvent({
    type: 'submit',
    ...overrides
  });

// Form validation helpers
export const fillFormFields = async (user, formData) => {
  const fields = Object.entries(formData);
  
  for (const [fieldName, value] of fields) {
    const input = document.querySelector(`input[name="${fieldName}"], select[name="${fieldName}"], textarea[name="${fieldName}"]`);
    if (input) {
      if (input.type === 'checkbox' || input.type === 'radio') {
        if (value) {
          await user.click(input);
        }
      } else {
        await user.type(input, String(value));
      }
    }
  }
};

// Assertion helpers
export const expectFormField = (container, fieldName, expectedValue = null) => {
  const field = container.querySelector(`[name="${fieldName}"]`);
  expect(field).toBeInTheDocument();
  
  if (expectedValue !== null) {
    if (field.type === 'checkbox' || field.type === 'radio') {
      expect(field.checked).toBe(Boolean(expectedValue));
    } else {
      expect(field.value).toBe(String(expectedValue));
    }
  }
  
  return field;
};

export const expectErrorMessage = (container, message) => {
  const errorElement = container.querySelector('[style*="background: #f8d7da"], [style*="color: #721c24"], .error-message');
  expect(errorElement).toBeInTheDocument();
  if (message) {
    expect(errorElement).toHaveTextContent(message);
  }
};

export const expectSuccessMessage = (container, message) => {
  const successElement = container.querySelector('[style*="background: #d4edda"], [style*="color: #155724"], .success-message');
  expect(successElement).toBeInTheDocument();
  if (message) {
    expect(successElement).toHaveTextContent(message);
  }
};

// Mock HTTP responses
export const mockApiResponse = (data, options = {}) => {
  const {
    status = 200,
    ok = true,
    delay = 0
  } = options;
  
  return new Promise(resolve => {
    setTimeout(() => {
      resolve({
        ok,
        status,
        statusText: ok ? 'OK' : 'Error',
        json: () => Promise.resolve(data),
        text: () => Promise.resolve(JSON.stringify(data))
      });
    }, delay);
  });
};

// Component test helpers
export const getByDataTestId = (container, testId) => {
  return container.querySelector(`[data-testid="${testId}"]`);
};

export const getAllByDataTestId = (container, testId) => {
  return container.querySelectorAll(`[data-testid="${testId}"]`);
};

// Form interaction patterns
export const submitForm = async (user, form) => {
  const submitButton = form.querySelector('button[type="submit"], input[type="submit"]');
  if (submitButton) {
    await user.click(submitButton);
  } else {
    // Submit via form if no submit button found
    await user.submit(form);
  }
};

export const selectSuggestedAmount = async (user, container, amount) => {
  const amountButton = container.querySelector(`button[type="button"]:has-text("$${amount}")`);
  if (amountButton) {
    await user.click(amountButton);
  }
  return amountButton;
};

// Wallet interaction helpers
export const connectWallet = async (user, container) => {
  const connectButton = container.querySelector('button:has-text("Connect"), button:has-text("MetaMask")');
  if (connectButton) {
    await user.click(connectButton);
  }
  return connectButton;
};

export const disconnectWallet = async (user, container) => {
  const disconnectButton = container.querySelector('button:has-text("Disconnect")');
  if (disconnectButton) {
    await user.click(disconnectButton);
  }
  return disconnectButton;
};

// Time utilities for testing
export const mockDateNow = (timestamp) => {
  const dateSpy = vi.spyOn(Date, 'now');
  dateSpy.mockReturnValue(timestamp);
  return dateSpy;
};

export const mockDate = (dateString) => {
  const mockDate = new Date(dateString);
  const dateSpy = vi.spyOn(global, 'Date');
  dateSpy.mockImplementation(() => mockDate);
  return dateSpy;
};

// Export everything for easy importing
export * from '@testing-library/react';
export * from '@testing-library/user-event';
export { vi } from 'vitest';