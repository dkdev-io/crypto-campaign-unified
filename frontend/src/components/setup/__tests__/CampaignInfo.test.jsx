import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render, screen, userEvent, waitFor } from '../../../utils/test-utils';
import CampaignInfo from '../CampaignInfo';

describe('CampaignInfo', () => {
  const defaultProps = {
    formData: {},
    updateFormData: vi.fn(),
    onNext: vi.fn(),
    onPrev: vi.fn()
  };

  beforeEach(() => {
    vi.clearAllMocks();
  });

  describe('Component Rendering', () => {
    it('renders campaign information component with title', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      expect(screen.getByText('Campaign Information')).toBeInTheDocument();
    });

    it('renders campaign name input field', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      expect(screen.getByText('Campaign Name')).toBeInTheDocument();
      expect(screen.getByRole('textbox', { name: /Campaign Name/i })).toBeInTheDocument();
    });

    it('renders website input field', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      expect(screen.getByText('Website')).toBeInTheDocument();
      expect(screen.getByRole('textbox', { name: /Website/i })).toBeInTheDocument();
    });

    it('renders navigation buttons', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      expect(screen.getByText('Back')).toBeInTheDocument();
      expect(screen.getByText('Next')).toBeInTheDocument();
    });
  });

  describe('Form Data Population', () => {
    it('displays existing campaign name from form data', () => {
      const propsWithData = {
        ...defaultProps,
        formData: {
          campaignName: 'Test Campaign 2024'
        }
      };

      render(<CampaignInfo {...propsWithData} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      expect(campaignNameInput.value).toBe('Test Campaign 2024');
    });

    it('displays existing website from form data', () => {
      const propsWithData = {
        ...defaultProps,
        formData: {
          website: 'https://campaign2024.com'
        }
      };

      render(<CampaignInfo {...propsWithData} />);
      
      const websiteInput = screen.getByRole('textbox', { name: /Website/i });
      expect(websiteInput.value).toBe('https://campaign2024.com');
    });

    it('shows empty fields when no form data provided', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      const websiteInput = screen.getByRole('textbox', { name: /Website/i });
      
      expect(campaignNameInput.value).toBe('');
      expect(websiteInput.value).toBe('');
    });
  });

  describe('User Input Handling', () => {
    let user;

    beforeEach(() => {
      user = userEvent.setup();
    });

    it('updates form data when campaign name is changed', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      await user.type(campaignNameInput, 'New Campaign Name');
      
      expect(defaultProps.updateFormData).toHaveBeenCalledWith({
        campaignName: 'New Campaign Name'
      });
    });

    it('updates form data when website is changed', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const websiteInput = screen.getByRole('textbox', { name: /Website/i });
      await user.type(websiteInput, 'https://newcampaign.com');
      
      expect(defaultProps.updateFormData).toHaveBeenCalledWith({
        website: 'https://newcampaign.com'
      });
    });

    it('calls updateFormData for each character typed', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      await user.type(campaignNameInput, 'Test');
      
      // Should be called for each character
      expect(defaultProps.updateFormData).toHaveBeenCalledTimes(4);
      expect(defaultProps.updateFormData).toHaveBeenNthCalledWith(1, { campaignName: 'T' });
      expect(defaultProps.updateFormData).toHaveBeenNthCalledWith(2, { campaignName: 'Te' });
      expect(defaultProps.updateFormData).toHaveBeenNthCalledWith(3, { campaignName: 'Tes' });
      expect(defaultProps.updateFormData).toHaveBeenNthCalledWith(4, { campaignName: 'Test' });
    });

    it('handles clearing input fields', async () => {
      const propsWithData = {
        ...defaultProps,
        formData: {
          campaignName: 'Original Name',
          website: 'https://original.com'
        }
      };

      render(<CampaignInfo {...propsWithData} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      await user.clear(campaignNameInput);
      
      expect(defaultProps.updateFormData).toHaveBeenCalledWith({
        campaignName: ''
      });
    });
  });

  describe('Navigation', () => {
    let user;

    beforeEach(() => {
      user = userEvent.setup();
    });

    it('calls onPrev when back button is clicked', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const backButton = screen.getByText('Back');
      await user.click(backButton);
      
      expect(defaultProps.onPrev).toHaveBeenCalledTimes(1);
    });

    it('calls onNext when next button is clicked', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const nextButton = screen.getByText('Next');
      await user.click(nextButton);
      
      expect(defaultProps.onNext).toHaveBeenCalledTimes(1);
    });

    it('allows navigation without form validation', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      // Should be able to click next even with empty fields
      const nextButton = screen.getByText('Next');
      await user.click(nextButton);
      
      expect(defaultProps.onNext).toHaveBeenCalled();
    });
  });

  describe('CSS Classes', () => {
    it('applies correct CSS classes to form elements', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      const websiteInput = screen.getByRole('textbox', { name: /Website/i });
      
      expect(campaignNameInput).toHaveClass('form-input');
      expect(websiteInput).toHaveClass('form-input');
    });

    it('applies correct CSS classes to form groups and actions', () => {
      const { container } = render(<CampaignInfo {...defaultProps} />);
      
      expect(container.querySelector('.form-group')).toBeInTheDocument();
      expect(container.querySelector('.form-actions')).toBeInTheDocument();
    });

    it('applies correct CSS classes to buttons', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const backButton = screen.getByText('Back');
      const nextButton = screen.getByText('Next');
      
      expect(backButton).toHaveClass('btn', 'btn-secondary');
      expect(nextButton).toHaveClass('btn', 'btn-primary');
    });
  });

  describe('Form Field Behavior', () => {
    let user;

    beforeEach(() => {
      user = userEvent.setup();
    });

    it('accepts text input in campaign name field', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      await user.type(campaignNameInput, 'My Campaign 2024');
      
      expect(campaignNameInput.value).toBe('My Campaign 2024');
    });

    it('accepts URL input in website field', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const websiteInput = screen.getByRole('textbox', { name: /Website/i });
      await user.type(websiteInput, 'https://mycampaign.org');
      
      expect(websiteInput.value).toBe('https://mycampaign.org');
    });

    it('handles special characters in inputs', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      await user.type(campaignNameInput, 'Campaign & Progress 2024!');
      
      expect(campaignNameInput.value).toBe('Campaign & Progress 2024!');
    });

    it('handles long input values', async () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const longCampaignName = 'A Very Long Campaign Name That Exceeds Normal Length Expectations For Testing Purposes';
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      await user.type(campaignNameInput, longCampaignName);
      
      expect(campaignNameInput.value).toBe(longCampaignName);
    });
  });

  describe('Props Integration', () => {
    it('correctly passes through all required props', () => {
      const customProps = {
        formData: {
          campaignName: 'Test Campaign',
          website: 'https://test.com'
        },
        updateFormData: vi.fn(),
        onNext: vi.fn(),
        onPrev: vi.fn()
      };

      render(<CampaignInfo {...customProps} />);
      
      // Component should render without errors and use the props
      expect(screen.getByDisplayValue('Test Campaign')).toBeInTheDocument();
      expect(screen.getByDisplayValue('https://test.com')).toBeInTheDocument();
    });

    it('handles missing form data properties gracefully', () => {
      const propsWithPartialData = {
        ...defaultProps,
        formData: {
          campaignName: 'Only Name Provided'
          // website is missing
        }
      };

      render(<CampaignInfo {...propsWithPartialData} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      const websiteInput = screen.getByRole('textbox', { name: /Website/i });
      
      expect(campaignNameInput.value).toBe('Only Name Provided');
      expect(websiteInput.value).toBe(''); // Should default to empty string
    });
  });

  describe('Accessibility', () => {
    it('has proper form labels', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      expect(screen.getByLabelText(/Campaign Name/i)).toBeInTheDocument();
      expect(screen.getByLabelText(/Website/i)).toBeInTheDocument();
    });

    it('has proper heading structure', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const heading = screen.getByRole('heading', { level: 2 });
      expect(heading).toHaveTextContent('Campaign Information');
    });

    it('has accessible buttons', () => {
      render(<CampaignInfo {...defaultProps} />);
      
      const backButton = screen.getByRole('button', { name: /Back/i });
      const nextButton = screen.getByRole('button', { name: /Next/i });
      
      expect(backButton).toBeInTheDocument();
      expect(nextButton).toBeInTheDocument();
    });

    it('maintains focus when typing in inputs', async () => {
      const user = userEvent.setup();
      render(<CampaignInfo {...defaultProps} />);
      
      const campaignNameInput = screen.getByRole('textbox', { name: /Campaign Name/i });
      await user.click(campaignNameInput);
      await user.type(campaignNameInput, 'Test');
      
      expect(campaignNameInput).toHaveFocus();
    });
  });

  describe('Component Snapshots', () => {
    it('matches snapshot for empty form', () => {
      const { container } = render(<CampaignInfo {...defaultProps} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('matches snapshot for populated form', () => {
      const propsWithData = {
        ...defaultProps,
        formData: {
          campaignName: 'Test Campaign 2024',
          website: 'https://test-campaign.org'
        }
      };

      const { container } = render(<CampaignInfo {...propsWithData} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});