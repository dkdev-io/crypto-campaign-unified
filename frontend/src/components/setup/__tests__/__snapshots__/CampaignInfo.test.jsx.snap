// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CampaignInfo > Component Snapshots > matches snapshot for empty form 1`] = `
<div>
  <h2>
    Campaign Information
  </h2>
  <div
    class="form-group"
  >
    <label>
      Campaign Name
    </label>
    <input
      class="form-input"
      value=""
    />
  </div>
  <div
    class="form-group"
  >
    <label>
      Website
    </label>
    <input
      class="form-input"
      value=""
    />
  </div>
  <div
    class="form-actions"
  >
    <button
      class="btn btn-secondary"
    >
      Back
    </button>
    <button
      class="btn btn-primary"
    >
      Next
    </button>
  </div>
</div>
`;

exports[`CampaignInfo > Component Snapshots > matches snapshot for populated form 1`] = `
<div>
  <h2>
    Campaign Information
  </h2>
  <div
    class="form-group"
  >
    <label>
      Campaign Name
    </label>
    <input
      class="form-input"
      value="Test Campaign 2024"
    />
  </div>
  <div
    class="form-group"
  >
    <label>
      Website
    </label>
    <input
      class="form-input"
      value="https://test-campaign.org"
    />
  </div>
  <div
    class="form-actions"
  >
    <button
      class="btn btn-secondary"
    >
      Back
    </button>
    <button
      class="btn btn-primary"
    >
      Next
    </button>
  </div>
</div>
`;
